cpp/src/arrow/array/array_test.cc

```diff
@@ -2162,6 +2162,16 @@ TEST_F(TestFWBinaryArray, ArrayDataVisitorSliced) {
   ARROW_UNUSED(visitor);  // Workaround weird MSVC warning
 }
 
+TEST_F(TestFWBinaryArray, ArrayIndexOperator) {
+  auto type = fixed_size_binary(3);
+  auto arr = ArrayFromJSON(type, R"(["abc", null, "def"])");
+  auto fsba = std::static_pointer_cast<FixedSizeBinaryArray>(arr);
```

Comment:
Please use `checked_pointer_cast` from `arrow/util/checked_cast.h`.

Reply:
yeah, my bad, there is no checked_pointer_cast in the other project I am working on, habitually I couldn't think of this one here.

Reply:
Did you forget to do this? :-)

Reply:
yes, hehe. Will push in next update, thank you